src/Bicep.Core.IntegrationTests/ScenarioTests.cs

```diff
@@ -6142,4 +6142,87 @@ public void Test_Issue14059_repro3()
 
         result.ExcludingLinterDiagnostics().Should().NotHaveAnyDiagnostics();
     }
+
+    [TestMethod]
+    public void Test_Issue2163_Deployments_Secure_Outputs_E2E()
```

Comment:
[nit] Would you mind creating a dedicated new test class for the secure outputs functionality?

Reply:
Will do, created a new file for it. Thanks!